### PR TITLE
:book: update the docs over regards book-v3 branch

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**
- Update the docs to have the info over the latest branch that tough to be used to v3

**Motivation**
- Closes : https://github.com/kubernetes-sigs/kubebuilder/issues/1860

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -95,8 +95,8 @@ separately.
 
 The docs are published off of three branches:
 
-- `book-v2`: [book.kubebuilder.io](https://book.kubebuilder.io) -- current
-  docs
+- `book-v3`: [book.kubebuilder.io](https://book.kubebuilder.io) -- current docs
+- `book-v2`: [book-v2.kubebuilder.io](https://book.kubebuilder.io) -- legacy docs
```
Comment (estroz) [inline CONTRIBUTING.md:99]: This PR should probably wait until a `book-v2` subdomain is created. @DirectXMan12 how would we go about doing that?

**Review by Adirio — APPROVED** [review]
Waiting on the comment above, the rest LGTM

File: VERSIONING.md [hunk 2]
```diff
@@ -43,7 +43,7 @@ The cherry-picks will automatically be published to the book once their PR
 merges.
 
 **When you publish a KubeBuilder release**, be sure to also submit a PR
-that merges the main branch into [book-v2][book-branch], so that it
+that merges the main branch into [book-v3][book-branch], so that it
 describes the latest changes in the new release.
 
 [book-branch]: https://github.com/kubernetes-sigs/kubebuilder/tree/tools-releases
```
Comment (Adirio) [inline VERSIONING.md:49]: This link is not right, it should be:
```suggestion
[book-branch]: https://github.com/kubernetes-sigs/kubebuilder/tree/book-v3
```

**Comment by Adirio** [comment]
> Motivation
> 
> - #1860

Please change it for:

> Closes: #1860

So that @k8s-ci-robot closes the issue when done.

**Comment by Adirio** [comment]
Rebase and squash required

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -95,8 +95,8 @@ separately.
 
 The docs are published off of three branches:
 
-- `book-v2`: [book.kubebuilder.io](https://book.kubebuilder.io) -- current
-  docs
+- `book-v3`: [book.kubebuilder.io](https://book.kubebuilder.io) -- current docs
+- `book-v2`: [book-v2.kubebuilder.io](https://book-v2.kubebuilder.io) -- legacy docs
```
Comment (estroz) [inline CONTRIBUTING.md:99]: This subdomain does not exist yet. I think a netlify mapping is needed.

**Review by estroz — APPROVED** [review]
/approve
/lgtm

**Pull request merged**